src/aspire/operators/filters.py

```diff
@@ -337,7 +349,8 @@ def _evaluate(self, omega):
         # Result is 1 x np.prod(self.sz) in shape; convert to a 1-d vector
         result = np.squeeze(result, 0)
 
-        return result
+        # Recast result with correct dtype
+        return result.astype(self.xfer_fn_array.dtype)
```

Comment:
See other remark, but if we still need this, lets make it `copy=False`. Thanks

Reply:
We still need this (see comment below). I will make it `copy=False`. Thanks.